jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -6614,36 +6618,175 @@ private static char convEndian(boolean isBigEndian, char value) {
 	 */
 	public native long getObjectSize(Object o);
 
+	public native boolean isClassFlattened(Class<?> clz);
+
+	private final <V> boolean compareAndSetFlattenedValue(Object obj, long offset, Class<?> clz, V v1, V v2) {
```

Comment:
Please add Doxygen comment for the new methods added.

Reply:
added in https://github.com/eclipse-openj9/openj9/commit/c54f1c95bd07867c80bf9bdc9e882849927b4c93

Reply:
is `compareAndSetFlattenedValue` ever used without the lock? If not then we can just and the lock inside the call

Reply:
Looks like it's not anymore. Also since it's only called by one function now (`compareAndSetValue`) I can just get rid of `compareAndSetFlattenedValue` completely and move its logic into `compareAndSetValue`